utils/dgcClient.ts

```diff
@@ -0,0 +1,31 @@
+import { NonEmptyString } from "@pagopa/ts-commons/lib/strings";
+import { Client as IDGCClient, createClient } from "../generated/dgc/client";
+import { IConfig } from "./config";
+import { sha256 } from "./conversions";
+import { DGCEnvironment, getFetchWithClientCertificate } from "./httpsAgent";
+
+export type IGetDGCClient = (fiscalCodeHash: NonEmptyString) => IDGCClient;
+
+export const getDGCClientBuilder: (
+  config: IConfig,
+  env: NodeJS.ProcessEnv
+) => IGetDGCClient = (config, env) => (
+  fiscalCodeHash: NonEmptyString
+): IDGCClient => {
+  if (config.DGC_LOAD_TEST_FISCAL_CODES.map(sha256).includes(fiscalCodeHash)) {
+    return createClient({
+      baseUrl: config.DGC_LOAD_TEST_URL.href,
+      fetchApi: getFetchWithClientCertificate(env, DGCEnvironment.LOAD_TEST)
+    });
+  }
+  if (config.DGC_UAT_FISCAL_CODES.map(sha256).includes(fiscalCodeHash)) {
+    return createClient({
+      baseUrl: config.DGC_UAT_URL.href,
+      fetchApi: getFetchWithClientCertificate(env, DGCEnvironment.UAT)
+    });
+  }
+  return createClient({
+    baseUrl: config.DGC_PROD_URL.href,
+    fetchApi: getFetchWithClientCertificate(env, DGCEnvironment.PROD)
+  });
+};
```

Comment:
I think this is in conflict with this other implementation:
https://github.com/pagopa/io-functions-eucovidcerts/pull/13/files#diff-07e1e85f45d882e2b350d1b1c590a599027c122b9deb3f414c01eda04d89554a
Maybe we have to merge them in the same function.

Reply:
Sorry I've missed that PR. Yes we need only one implementation, both do the same thing in different ways